### DISCUSSION TITLE
Intel NUC (Mini PC) builtin Bluetooth and USB Bluetooth Dongle Problems, TIP: use ESP32 Bluetooth Proxy

### DISCUSSION THREAD

**Discussion by ItsMe00007** [Patrick762/hassio-bluetti-bt#78] [General]
With ESP32 Bluetooth Proxy my AC500 was automatically recognised and configured from the Bluetti-BT integration.

But since my Intel Mini-PC has got a Built-in Bluetooth adapter I disabled the Bluetooth Proxy and enabled the builtin Bluetooth adapter.
Problems:
1. The Bluetti-bt integration showed a list of Bluetooth MAC-Adresses, but my AC500 was not in the list. 
   But after a few Minutes the already configured integration from the ESP32 connection was filled with values and it startet to work.
  
2. To calibrate the BMS i discharged both batteries to 0 and my AC500 turned off in the early morning. 
     So the AC500 was down and not reachable via Bluetooth for a few hours.
     When the sun came up the PV startet the AC500 startet to work again.
     Unfortunately the builtin Bluetooth Adapter was not available anymore in HA.
     Rebooting HA completely did not help, also manual shutdown and manual Power on did not help to bring the Bluetooth      adapter back.
     Only after shutting down and DISCONNECTING the Power Supply and after CONNECTING the Power-Supply back brought      the builtin Intel Bluetooth adapter back to live.
     I had that Problem also in the past with other bluetooth integrations, also with another Bluetooth adapter from ASUS.  
     
     It seems that the HA OS has Problems with builtin Bluetooth or USB Bluetooth Dongle if there are no responses from      Bluetooth device.
     According to the HA Core Log timeout when trying to reset the adapter.
     
     I therefore recommend using an ESP32 Bluetooth Proxy for the Bluetti_BT integration..
     
     How is your experience with the HA Bluetooth integration ?

**Comment by Patrick762** [comment]
I only use the BT Proxy. Makes it much easier to reach every device

**Reply by ItsMe00007** [reply]
I moved now from WIFI BT Proxy to Ethernet attached BT Proxy using the config from https://github.com/esphome/firmware/blob/main/bluetooth-proxy/wt32-eth01.yaml and an wt32-eth01 board + an USB-2-UART Adapter.
According to a tip from the ESPHome Discord it could be necessary to add the following to the .yaml file to switch the flash mode to "die"

platformio_options:
    board_build.flash_mode: dio
    
    
That is more stable then a WIFI BT Proxy.

**Reply by Patrick762** [reply]
Could explain many problems I have right now
